templates/common/on-prem/files/coredns-corefile.yaml

```diff
@@ -5,7 +5,6 @@ contents:
     . {
         errors
         health :18080
-        mdns {{ .DNS.Spec.BaseDomain }} 0 {{`{{.Cluster.Name}}`}} {{`{{.NonVirtualIP}}`}}
```

Comment:
Seems like we no longer need the rule for mdns in the security groups and it should be removed from the installer. Are we tracking this somewhere?

Reply:
Ah, no I hadn't considered that because baremetal doesn't have security groups. What would be the best way to make sure that doesn't get lost?

Reply:
Normally, I would say in the same BZ that tracks the removal of mdns, or just a patch in the installer, but with only 3 days to feature freeze I think it's unrealistic to expect both patch to merge before FF. I'll create the BZ.

Reply:
We're now tracking mdns removal in https://bugzilla.redhat.com/show_bug.cgi?id=1946506. Aside from allowing to merge past feature freeze, this also ensures QE looks at and verifies the code change.

Reply:
Do we still need mdns-publisher?

Reply:
As it stands right now, we do. My thought was that after this and https://github.com/openshift/machine-config-operator/pull/2410 merged we could remove mdns-publisher completely, but right now some platforms are still going to be using mdns.

Given how close we are to feature freeze now, maybe I need to pull 2410 into this PR and do everything at once.